### ISSUE TITLE
byte array throws a syntax error when inserting into postgresql

### ISSUE THREAD

**Issue by willprice** [schambers/fluentmigrator#630]
```csharp
Insert.IntoTable("file").Row(new
{
     id = 1,
     contents = someByteArray
}
```

Causes the error
```

!!! An error occurred executing the following sql:
INSERT INTO "public"."file" ("id","contents") VALUES (1,0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000);
The error was ERROR: 42601: syntax error at or near "x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"

!!! ERROR: 42601: syntax error at or near "x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
```
(my byte array is a 100 item long array of zeros)

**Comment by tommarien** [comment]
@willprice Can you try it with manual execute.sql

INSERT INTO "public"."file" ("id","contents") VALUES (1,"\\x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000");

I am guessing that the genericquoter formatByteArray is the culprit

**Comment by willprice** [comment]
That too throws the same error, the [postgresql docs](http://www.postgresql.org/docs/9.4/static/datatype-binary.html) show how to escape byte arrays. You use the format `E'\\x<data>'`, perhaps override `FormatByteData` in `PostgresQuoter` with something like
```csharp
@"E'\\x" + string.Concat(data.Select(b => b.ToString("X2")).ToArray()) + "'"
```

I'm currently using that as my workaround and calling `RawSql.Insert`.

**Comment by tommarien** [comment]
@willprice That will be according to spec, i'll mark this as an up for graps issue, so maybe some good person can do the fix ;)

**Comment by willprice** [comment]
Thanks for the tips on where to look on solving this! Let me know if you don't want a dependency on LINQ, if so, I'll do it without.

**Comment by tommarien** [comment]
@willprice No the linq dependency is fine ;) thanks for the pull request ;)

**Comment by dealproc** [comment]
ha.  this just bit me tonight.  was getting ready to build a PR for it.  thanks @willprice :)  I'll do an ado.net type migration for now, but will factor over to this once the next release comes out.

**Comment by tommarien** [comment]
closed by merging #631
